File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> jumpsTo(n, false);
        jumpsTo[n-1] = true;
        for(int i = n-2; i >= 0; i--){
            for(auto j = 1; j <= nums[i]; j++){
                if(jumpsTo[i+j] == true){
                    jumpsTo[i] = true;
                    break;
                }
            }
        }
        return jumpsTo[0];
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        resultArr = list()
        nums.sort()
        firstIdx = 0
        while (firstIdx < len(nums) - 2):
            if nums[firstIdx] > 0:
                break
            if firstIdx > 0 and nums[firstIdx] == nums[firstIdx - 1]:
                firstIdx += 1
                continue
            secondIdx = firstIdx + 1
            thirdIdx = len(nums) - 1
            while (secondIdx < thirdIdx):
                sum = nums[firstIdx] + nums[secondIdx] + nums[thirdIdx]
                if sum == 0:
                    resultArr.append([nums[firstIdx], nums[secondIdx], nums[thirdIdx]])
                    secondIdx, thirdIdx = secondIdx + 1, thirdIdx -1
                    while(thirdIdx > secondIdx and nums[thirdIdx] == nums[thirdIdx+1]):
                        thirdIdx -= 1
                elif sum > 0:
                    thirdIdx -= 1
                else:
                    secondIdx += 1
            firstIdx += 1
        return resultArr
        
        

        